--- a/kernel/nv-drm.c	2014-09-11 23:33:06.000000000 +0100
+++ b/kernel/nv-drm.c	2014-10-21 12:12:59.472379677 +0100
@@ -17,6 +17,9 @@
 #if defined(NV_DRM_AVAILABLE)
 
 #include <drm/drmP.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0)
+#include <drm/drm_gem.h>
+#endif
 
 extern nv_linux_state_t *nv_linux_devices;
 
@@ -124,6 +127,10 @@ static struct drm_driver nv_drm_driver =
     .gem_prime_vmap = nv_gem_prime_vmap,
     .gem_prime_vunmap = nv_gem_prime_vunmap,
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0)
+    .set_busid = drm_pci_set_busid,
+#endif
+
     .name = "nvidia-drm",
     .desc = "NVIDIA DRM driver",
     .date = "20130102",
