diff -uNrp a/kernel/nvidia-uvm/uvm_common.c b/kernel/nvidia-uvm/uvm_common.c
--- a/kernel/nvidia-uvm/uvm_common.c	2017-09-20 02:33:10.000000000 +0100
+++ b/kernel/nvidia-uvm/uvm_common.c	2017-09-23 09:02:44.989610766 +0100
@@ -388,5 +388,9 @@ module_param(uvm_enable_builtin_tests, i
 MODULE_PARM_DESC(uvm_enable_builtin_tests,
                  "Enable the UVM built-in tests. (This is a security risk)");
 
+#if LINUX_VERSION_CODE > KERNEL_VERSION(4, 13, 0)
+MODULE_LICENSE("Dual MIT/GPL");
+#else
 MODULE_LICENSE("MIT");
+#endif
 MODULE_INFO(supported, "external");
