--- a/kernel/nvidia/nv-procfs.c
+++ b/kernel/nvidia/nv-procfs.c
@@ -359,8 +359,12 @@
         nvl = NV_GET_NVL_FROM_NV_STATE(nv);
     registry_keys = ((nvl != NULL) ?
             nvl->registry_keys : nv_registry_keys);
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,3,0)
+    seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    return 0;
+#else
     return seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+#endif
 }
 
 static ssize_t
@@ -560,7 +564,12 @@
     void *v
 )
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,3,0)
+    seq_puts(s, s->private);
+    return 0;
+#else
     return seq_puts(s, s->private);
+#endif
 }
 
 NV_DEFINE_PROCFS_SINGLE_FILE(text_file);
