--- a/kernel/nv-i2c.c
+++ b/kernel/nv-i2c.c
@@ -324,13 +324,10 @@ BOOL NV_API_CALL nv_i2c_del_adapter(nv_state_t *nv, void *data)
     if (!pI2cAdapter) return FALSE;
 
     // attempt release with the OS
-    osstatus = i2c_del_adapter(pI2cAdapter);
+    i2c_del_adapter(pI2cAdapter);
 
-    if (!osstatus)
-    {
-        os_free_mem(pI2cAdapter);
-        wasReleased = TRUE;
-    }
+	os_free_mem(pI2cAdapter);
+	wasReleased = TRUE;
 
     return wasReleased;
 }
--- a/kernel/nv-procfs.c
+++ b/kernel/nv-procfs.c
@@ -14,6 +14,8 @@
 #include "os-interface.h"
 #include "nv-linux.h"
 
+#undef CONFIG_PROC_FS
+
 #if defined(CONFIG_PROC_FS)
 
 #include "nv_compiler.h"
--- a/kernel/nv-linux.h
+++ b/kernel/nv-linux.h
@@ -957,7 +957,11 @@ static inline int nv_execute_on_all_cpus
 #endif
 
 #if !defined(NV_VMWARE)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
+#define NV_NUM_PHYSPAGES                get_num_physpages()
+#else
 #define NV_NUM_PHYSPAGES                num_physpages
+#endif
 #define NV_GET_CURRENT_PROCESS()        current->tgid
 #define NV_IN_ATOMIC()                  in_atomic()
 #define NV_LOCAL_BH_DISABLE()           local_bh_disable()
