diff -uNrp a/conftest.sh b/conftest.sh
--- a/conftest.sh	2024-07-30 21:14:40.000000000 +0100
+++ b/conftest.sh	2024-08-17 12:15:07.311710767 +0100
@@ -6610,6 +6610,25 @@ compile_test() {
             compile_check_conftest "$CODE" "NV_DRM_FBDEV_GENERIC_SETUP_PRESENT" "" "functions"
         ;;
 
+        drm_fbdev_ttm_setup)
+            #
+            # Determine whether drm_fbdev_ttm_setup is present.
+            #
+            # Added by commit 9060d7f49376 ("drm/fb-helper: Finish the
+            # generic fbdev emulation") in v4.19.
+            #
+            CODE="
+            #include <drm/drm_fb_helper.h>
+            #if defined(NV_DRM_DRM_FBDEV_TTM_H_PRESENT)
+            #include <drm/drm_fbdev_ttm.h>
+            #endif
+            void conftest_drm_fbdev_ttm_setup(void) {
+                drm_fbdev_ttm_setup();
+            }"
+
+            compile_check_conftest "$CODE" "NV_DRM_FBDEV_TTM_SETUP_PRESENT" "" "functions"
+        ;;
+
         drm_aperture_remove_conflicting_pci_framebuffers)
             #
             # Determine whether drm_aperture_remove_conflicting_pci_framebuffers is present.
diff -uNrp a/header-presence-tests.mk b/header-presence-tests.mk
--- a/header-presence-tests.mk	2024-07-30 21:14:40.000000000 +0100
+++ b/header-presence-tests.mk	2024-08-17 12:11:21.315334757 +0100
@@ -15,6 +15,7 @@ NV_HEADER_PRESENCE_TESTS = \
   drm/drm_atomic_uapi.h \
   drm/drm_drv.h \
   drm/drm_fbdev_generic.h \
+  drm/drm_fbdev_ttm.h \
   drm/drm_framebuffer.h \
   drm/drm_connector.h \
   drm/drm_probe_helper.h \
diff -uNrp a/nvidia-drm/nvidia-drm-drv.c b/nvidia-drm/nvidia-drm-drv.c
--- a/nvidia-drm/nvidia-drm-drv.c	2024-07-30 21:31:46.000000000 +0100
+++ b/nvidia-drm/nvidia-drm-drv.c	2024-08-17 15:56:09.777675388 +0100
@@ -73,8 +73,13 @@
 #include <drm/drm_fbdev_generic.h>
 #endif
 
+#if defined(NV_DRM_DRM_FBDEV_TTM_H_PRESENT)
+#include <drm/drm_fbdev_ttm.h>
+#endif
+
 #include <linux/pci.h>
 #include <linux/workqueue.h>
+#include <linux/version.h>
 
 /*
  * Commit fcd70cd36b9b ("drm: Split out drm_probe_helper.h")
@@ -1851,7 +1856,11 @@ void nv_drm_register_drm_device(const nv
             drm_aperture_remove_conflicting_pci_framebuffers(pdev, nv_drm_driver.name);
 #endif
         }
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
         drm_fbdev_generic_setup(dev, 32);
+#else
+        drm_fbdev_ttm_setup(dev, 32);
+#endif
     }
 #endif /* defined(NV_DRM_FBDEV_GENERIC_AVAILABLE) */
 
diff -uNrp a/nvidia-drm/nvidia-drm-os-interface.h b/nvidia-drm/nvidia-drm-os-interface.h
--- a/nvidia-drm/nvidia-drm-os-interface.h	2024-07-30 21:31:46.000000000 +0100
+++ b/nvidia-drm/nvidia-drm-os-interface.h	2024-08-17 15:49:18.703671076 +0100
@@ -58,7 +58,7 @@ typedef struct nv_timer nv_drm_timer;
 #error "Need to define kernel timer callback primitives for this OS"
 #endif
 
-#if defined(NV_DRM_FBDEV_GENERIC_SETUP_PRESENT) && defined(NV_DRM_APERTURE_REMOVE_CONFLICTING_PCI_FRAMEBUFFERS_PRESENT)
+#if defined(NV_DRM_FBDEV_GENERIC_SETUP_PRESENT) || defined(NV_DRM_FBDEV_TTM_SETUP_PRESENT) && defined(NV_DRM_APERTURE_REMOVE_CONFLICTING_PCI_FRAMEBUFFERS_PRESENT)
 #define NV_DRM_FBDEV_GENERIC_AVAILABLE
 #endif
 
diff -uNrp a/nvidia-drm/nvidia-drm-sources.mk b/nvidia-drm/nvidia-drm-sources.mk
--- a/nvidia-drm/nvidia-drm-sources.mk	2024-07-30 21:17:37.000000000 +0100
+++ b/nvidia-drm/nvidia-drm-sources.mk	2024-08-17 16:06:05.645589668 +0100
@@ -67,6 +67,7 @@ NV_CONFTEST_FUNCTION_COMPILE_TESTS += fe
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += sync_file_get_fence
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_aperture_remove_conflicting_pci_framebuffers
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_fbdev_generic_setup
+NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_fbdev_ttm_setup
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_connector_attach_hdr_output_metadata_property
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_helper_crtc_enable_color_mgmt
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_crtc_enable_color_mgmt
