--- a/kernel/nvidia-drm/nvidia-drm-drv.c
+++ b/kernel/nvidia-drm/nvidia-drm-drv.c
@@ -659,7 +659,12 @@
 
 static struct drm_driver nv_drm_driver = {
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
+    .driver_features        = DRIVER_GEM | DRIVER_RENDER,
+#else
     .driver_features        = DRIVER_GEM | DRIVER_PRIME | DRIVER_RENDER,
+#endif
 
     .gem_free_object        = nv_drm_gem_free,
 

