diff -up nvidiapkg-armv7hl/kernel/nv-linux.h.arm nvidiapkg-armv7hl/kernel/nv-linux.h
--- nvidiapkg-armv7hl/kernel/nv-linux.h.arm 2014-08-01 03:14:20.000000000 +0200
+++ nvidiapkg-armv7hl/kernel/nv-linux.h	2014-08-07 11:06:30.925997327 +0200
@@ -230,7 +230,8 @@ RM_STATUS nvos_forward_error_to_cray(str
  * NVIDIA graphics driver has no way of knowing and is unlikely
  * to work correctly.
  */
-#if defined(CONFIG_XEN) && !defined(CONFIG_PARAVIRT)
+#if defined(CONFIG_XEN) && !defined(CONFIG_PARAVIRT) && \
+    !defined(NVCPU_ARM)
 #include <asm/maddr.h>
 #include <xen/interface/memory.h>
 #define NV_XEN_SUPPORT_OLD_STYLE_KERNEL
