diff -up nvidiapkg-armv7hl/kernel/common/inc/nv-linux.h.orig nvidiapkg-armv7hl/kernel/common/inc/nv-linux.h
--- nvidiapkg-armv7hl/kernel/common/inc/nv-linux.h.orig	2015-11-17 03:04:13.000000000 +0100
+++ nvidiapkg-armv7hl/kernel/common/inc/nv-linux.h	2016-06-22 16:15:28.975692646 +0200
@@ -454,6 +454,7 @@ typedef spinlock_t                nv_spi
 #if defined(CONFIG_OUTER_CACHE)
 #define OUTER_FLUSH_RANGE(start, end)   outer_flush_range((start),(end))
 #endif
+#include <asm/barrier.h>
 #define WRITE_COMBINE_FLUSH()    { dsb(); outer_sync(); }
 #elif defined(NVCPU_AARCH64)
     static inline void nv_flush_cache_cpu(void *info)
