diff -up kernel-open/nvidia/linux_nvswitch.c.orig kernel-open/nvidia/linux_nvswitch.c
--- kernel-open/nvidia/linux_nvswitch.c.orig	2021-12-31 13:53:49.000000000 +0100
+++ kernel-open/nvidia/linux_nvswitch.c	2022-01-19 13:17:53.955709286 +0100
@@ -1320,6 +1320,9 @@ nvswitch_probe
 
     pci_set_master(pci_dev);
 
+// Don't call pci_request_regions if CONFIG_SYSFB_SIMPLEFB - see also
+// https://patchwork.kernel.org/project/dri-devel/patch/20220117180359.18114-1-zack@kde.org/#24702504
+#if ! defined(CONFIG_SYSFB_SIMPLEFB)
     rc = pci_request_regions(pci_dev, nvswitch_dev->name);
     if (rc)
     {
@@ -1328,6 +1331,7 @@ nvswitch_probe
                rc);
         goto pci_request_regions_failed;
     }
+#endif
 
     nvswitch_dev->bar0 = pci_iomap(pci_dev, 0, 0);
     if (!nvswitch_dev->bar0)
