diff -uNrp a/kernel/nvidia-drm/nvidia-drm-drv.c b/kernel/nvidia-drm/nvidia-drm-drv.c
--- a/kernel/nvidia-drm/nvidia-drm-drv.c	2025-01-24 20:50:25.000000000 +0000
+++ b/kernel/nvidia-drm/nvidia-drm-drv.c	2025-02-08 10:40:30.455105183 +0000
@@ -86,6 +86,7 @@
 #include <drm/drm_fbdev_generic.h>
 #endif
 
+#include <linux/version.h>
 #include <linux/pci.h>
 #include <linux/workqueue.h>
 
@@ -1915,7 +1916,9 @@ static struct drm_driver nv_drm_driver =
     .name                   = "nvidia-drm",
 
     .desc                   = "NVIDIA DRM driver",
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
     .date                   = "20160202",
+#endif
 
 #if defined(NV_DRM_DRIVER_HAS_DEVICE_LIST)
     .device_list            = LIST_HEAD_INIT(nv_drm_driver.device_list),
