diff -uNrp a/kernel/nvidia/nv-reg.h b/kernel/nvidia/nv-reg.h
--- a/kernel/nvidia/nv-reg.h	2026-09-05 00:27:59.000000000 +0100
+++ b/kernel/nvidia/nv-reg.h	2026-09-10 06:51:23.515663470 +0100
@@ -1057,8 +1057,7 @@ NV_DEFINE_REG_ENTRY(__NV_ENABLE_PCIE_GEN
 NV_DEFINE_REG_ENTRY(__NV_ENABLE_MSI, 1);
 NV_DEFINE_REG_ENTRY(__NV_ENABLE_STREAM_MEMOPS, 0);
 NV_DEFINE_REG_ENTRY(__NV_RM_PROFILING_ADMIN_ONLY_PARAMETER, 1);
-NV_DEFINE_REG_ENTRY(__NV_PRESERVE_VIDEO_MEMORY_ALLOCATIONS,
-                    NV_REG_PRESERVE_VIDEO_MEMORY_ALLOCATIONS_AUTO);
+NV_DEFINE_REG_ENTRY(__NV_PRESERVE_VIDEO_MEMORY_ALLOCATIONS, 1);
 NV_DEFINE_REG_ENTRY(__NV_ENABLE_S0IX_POWER_MANAGEMENT, 0);
 NV_DEFINE_REG_ENTRY(__NV_S0IX_POWER_MANAGEMENT_VIDEO_MEMORY_THRESHOLD, 256);
 NV_DEFINE_REG_ENTRY(__NV_DYNAMIC_POWER_MANAGEMENT, 3);
diff -uNrp a/kernel/nvidia/os-interface.c b/kernel/nvidia/os-interface.c
--- a/kernel/nvidia/os-interface.c	2026-09-05 00:27:53.000000000 +0100
+++ b/kernel/nvidia/os-interface.c	2026-09-10 06:52:45.421338188 +0100
@@ -1889,7 +1889,7 @@ NV_STATUS NV_API_CALL os_allocate_tempor
 {
     struct file *file = ERR_PTR(-EOPNOTSUPP);
 #if NV_FILESYSTEM_ACCESS_AVAILABLE
-    const char *default_path = "/tmp";
+    const char *default_path = "/var/tmp";
     const int flags = O_TMPFILE | O_LARGEFILE | O_RDWR;
     const char *path = NVreg_TemporaryFilePath;
 

diff -uNrp a/kernel-open/nvidia/os-interface.c b/kernel-open/nvidia/os-interface.c
--- a/kernel-open/nvidia/os-interface.c	2026-09-04 23:33:41.000000000 +0100
+++ b/kernel-open/nvidia/os-interface.c	2026-09-10 06:39:56.650284848 +0100
@@ -1985,7 +1985,7 @@ NV_STATUS NV_API_CALL os_allocate_tempor
 {
     struct file *file = ERR_PTR(-EOPNOTSUPP);
 #if NV_FILESYSTEM_ACCESS_AVAILABLE
-    const char *default_path = "/tmp";
+    const char *default_path = "/var/lib/systemd/sleep";
     const int flags = O_TMPFILE | O_LARGEFILE | O_RDWR;
     const char *path = NVreg_TemporaryFilePath;
 
